tests/std/tests/P0896R4_counted_iterator/test.cpp

```diff
@@ -0,0 +1,286 @@
+            assert(*constIter == 1);
+            if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                assert(addressof(*constIter) == input);
+            }
+
+            if constexpr (random_access_iterator<Iter>) {
+                assert(iter[1] == 2);
+                if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                    assert(addressof(iter[1]) == input + 1);
+                }
+            }
+        }
+        // [counted.iter.nav]
+        {
+            { // pre increment
+                counted_iterator<Iter> iter{Iter{input}, 2};
+                assert(addressof(++iter) == addressof(iter));
+                assert(iter.count() == 1);
+                assert(*iter == 2);
+                if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                    assert(addressof(*iter) == input + 1);
+                }
+            }
+            if constexpr (forward_iterator<Iter>) { // post increment
+                counted_iterator<Iter> iter{Iter{input}, 2};
+                auto ref   = iter;
+                auto iter2 = iter++;
+                assert(iter2 == ref);
+                assert(iter.count() == 1);
+                assert(*iter == 2);
+                if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                    assert(addressof(*iter) == input + 1);
+                }
+            }
+            if constexpr (bidirectional_iterator<Iter>) {
+                {
+                    // pre decrement
+                    counted_iterator<Iter> iter{Iter{input + 2}, 3};
+                    assert(addressof(--iter) == addressof(iter));
+                    assert(iter.count() == 4);
+                    assert(*iter == 2);
+                    if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                        assert(addressof(*iter) == input + 1);
+                    }
+                }
+                { // post decrement
+                    counted_iterator<Iter> iter{Iter{begin(input) + 2}, 3};
+                    auto ref   = iter;
+                    auto iter2 = iter--;
+                    assert(iter2 == ref);
+                    assert(iter.count() == 4);
+                    assert(*iter == 2);
+                    if constexpr (is_reference_v<iter_reference_t<Iter>>) {
+                        assert(addressof(*iter) == input + 1);
+                    }
+                }
+            }
+            if constexpr (random_access_iterator<Iter>) {
+                { // increment by n
+                    counted_iterator<Iter> iter{Iter{input}, 5};
+                    auto ref = iter + 3;
+                    assert(iter.count() == 5);
+                    assert(*iter == 1);
+                    assert(iter.count() == ref.count() + 3);
+                    assert(iter.base().peek() == prev(ref.base().peek(), 3));
+                }
+                { // post increment by n

```

Comment:
I think that describing `3 + iter` as "post increment" is confusing. It's not like `iter++` at all.

Reply:
Changed to increment by n with lhs / rhs to indicate why we have different clauses